### PR TITLE
Generator identifications

### PR DESCRIPTION
**Pull request by regular-citizen**
This PR introduces new UI for generator identifications.

### CODE REVIEW

**Comment by regular-citizen** [comment]
Thanks for the patch. I'll see if we can merge tomorrow.

**Comment by regular-citizen** [comment]
@calintat you patch seems to make a test fail. I think you framing removal logic is wrong. I'll patch it later, but it's quite worrying that CI did not pick it up.

**Comment by calintat** [comment]
Thanks, I just fixed it.

File: homotopy-core/src/rewrite.rs [hunk 1]
```diff
@@ -377,7 +377,7 @@ impl Rewrite0 {
                 target.replace(from, to),
                 label
                     .as_ref()
-                    .filter(|_| oriented && (target.generator == from || target.generator == to))
+                    .filter(|_| !oriented || (target.generator != to && target.generator != from))
```
Comment (calintat) [inline homotopy-core/src/rewrite.rs:380]: Ooops! Thanks for fixing this!

**Pull request merged**